### ISSUE TITLE
The new rules dialog in Outlook that can help set some rules for maintaining messages is not accesible completely with NVDA.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3069]
**Reported by sumandogra on 2013-03-09 15:40**
Outlook 2010
The new rules dialog in Outlook needs to be made more accessible.  The stop 2 in the dialog that helps defining the parameters depending on the underlined values cannot be set as one does not come to know what to do next after selecting the conditions.  No info is give b NVDA about the underlined info to be set or any such info that can help set a new rule.
Steps to reproduce:
1.  In Outlook, go to home tab>rules>manage rules and alerts.
2.  Rules and alerts dialog opens.
3.  Select new rule button.
4.  Follow the wizard and try to set a new rule with the help of NVDA.  But it becomes very difficult to set a rule with usual help form NVDA.

**Comment by ehollig** [comment]
I have not had an issue creating rules using Outlook 2016, although this could possibly be specific to Outlook 2010. Is there anyone with Outlook 2010 who could test using the manage rules and alerts and creating a new rule?

**Comment by Adriani90** [comment]
@Qchristensen did you not create a similar issue as well?

**Comment by Qchristensen** [comment]
I created #8597 which was that when you open the rules and alerts dialog and navigate through the rules, NVDA was not reading the currently selected rule correctly.  This is a different issue, and I think the argument could be made either way about whether it is an issue or not.

I don't currently have a copy of Office 2010, I do have Office 2007 and Office 365, and to be honest, the dialog presents much the same in both.  After choosing a rule template ("Move messages from someone to a folder" for instance), when you press TAB, the focus moves to "Step 2: Edit the rule description (click on an underlined value)".  Step 2 has a box which presents the rule almost as a sentence: "Apply this rule after the message arrives
from people or public groups
move it to the specified folder
and stop processing more rules"

In this case, "people or public groups" are underlined as a link, and so is "specified folder".  The first lets you choose who to filter messages from, and the second being which folder to put those messages in.

The issue is that when you press TAB to get to this part of the dialog, the focus jumps to the first underlined value - "people or public groups" and does not read any of the surrounding text to give it the context that a sighted user would be given.  It is certainly possible to complete creating a rule without that extra context.

It is also possible to read the additional context (after pressing TAB to move to "people or public groups" in this example):
- Press NVDA+numpad 4 (NVDA+shift+left arrow in laptop layout) to move to the previous object, which reads the text before the first link.
- Next, press NVDA+numpad 6 (NVDA+shift+right arrow in laptop layout) to move through the remainder of objects in this section.

So, I suggest there are two ways we can go with this issue:

1) Close the issue and call it a matter of learning the nuances of creating rules in Outlook and using object navigation to get additional information.

or

2) Implement an obvious way of NVDA reporting the information in that step 2 part of the dialog.  Perhaps when focus first moves to that section, where NVDA currently reads:
"Step 2: Edit the rule description (move with arrows and press SPACEBAR to edit parameters)  list
from people or public group"
to have it then go on to read:
"Apply this rule after the message arrives
from people or public group
with specific words in the subject
move it to the specified folder
stop processing more rules"

It is perhaps worth checking how easy that second option is.  If possible, and perhaps even if a quick fix, that could make the rules dialog easier to work with.

**Comment by Adriani90** [comment]
Thank you Quentin for your fast reply.

In my view we should use all ways of getting information which are provided by NVDA. Object navigation is accessible and this makes many dialogs accessible too. In my view, object navigation is the most powerful navigation tool on Windows. I think customizing every dialog to make it simpler is probably too much. But I guess we have to make object navigation more atractive to beginners. From my experience I can tell that there are many beginners who are not very open to this kind of navigation because it is too different for them compared to the usual way of navigation in Windows. And there are only few users who seriously learn and want to understand object navigation after some years of using NVDA.

 

I think we must fin sort of a compromise on this for the future.

 

 

Von: Quentin Christensen <notifications@github.com> 
Gesendet: Dienstag, 27. November 2018 23:41
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] The new rules dialog in Outlook that can help set some rules for maintaining messages is not accesible completely with NVDA. (#3069)

 

I created #8597 <https://github.com/nvaccess/nvda/issues/8597>  which was that when you open the rules and alerts dialog and navigate through the rules, NVDA was not reading the currently selected rule correctly. This is a different issue, and I think the argument could be made either way about whether it is an issue or not.

I don't currently have a copy of Office 2010, I do have Office 2007 and Office 365, and to be honest, the dialog presents much the same in both. After choosing a rule template ("Move messages from someone to a folder" for instance), when you press TAB, the focus moves to "Step 2: Edit the rule description (click on an underlined value)". Step 2 has a box which presents the rule almost as a sentence: "Apply this rule after the message arrives
from people or public groups
move it to the specified folder
and stop processing more rules"

In this case, "people or public groups" are underlined as a link, and so is "specified folder". The first lets you choose who to filter messages from, and the second being which folder to put those messages in.

The issue is that when you press TAB to get to this part of the dialog, the focus jumps to the first underlined value - "people or public groups" and does not read any of the surrounding text to give it the context that a sighted user would be given. It is certainly possible to complete creating a rule without that extra context.

It is also possible to read the additional context (after pressing TAB to move to "people or public groups" in this example):

*	Press NVDA+numpad 4 (NVDA+shift+left arrow in laptop layout) to move to the previous object, which reads the text before the first link.
*	Next, press NVDA+numpad 6 (NVDA+shift+right arrow in laptop layout) to move through the remainder of objects in this section.

So, I suggest there are two ways we can go with this issue:

1.	Close the issue and call it a matter of learning the nuances of creating rules in Outlook and using object navigation to get additional information.

or

2.	Implement an obvious way of NVDA reporting the information in that step 2 part of the dialog. Perhaps when focus first moves to that section, where NVDA currently reads:
"Step 2: Edit the rule description (move with arrows and press SPACEBAR to edit parameters) list
from people or public group"
to have it then go on to read:
"Apply this rule after the message arrives
from people or public group
with specific words in the subject
move it to the specified folder
stop processing more rules"

It is perhaps worth checking how easy that second option is. If possible, and perhaps even if a quick fix, that could make the rules dialog easier to work with.

**Comment by Adriani90** [comment]
This is still reproducible.

NVDA does not read 
"after the message arrives"
and
"stop processing more rules""
when landing on the step 2 list items with tab and shift+tab.

The problem is that these two parts are not list items, but are on the same object level as the list items as separate child objects. Actually they should be part of the list control property description, on the parent object level and not on the same object level as the list items.
Then probably NVDA could handle this via object description setting.

cc: @josephsl.

**Comment by Adriani90** [comment]
cc: @msftedad can you forward this to the development team please?
